src/dbt_osmosis/core/osmosis.py

```diff
@@ -314,34 +317,52 @@ def augment_existing_model(
     @lru_cache(maxsize=5000)
     def get_columns(self, parts: Tuple[str, str, str]) -> List[str]:
         """Get all columns in a list for a model"""
-        with self.adapter.connection_named("dbt-osmosis"):
-            table = self.adapter.get_relation(*parts)
-            columns = []
-            if not table:
-                logger().info(
-                    (
-                        ":cross_mark: Relation %s.%s.%s does not exist in target database, cannot"
-                        " resolve columns"
-                    ),
-                    *parts,
-                )
+
+        # If we provide a catalog, we read from it
+        if self.catalog_file:
+            file_path = Path(self.catalog_file)
+
+            content = json.loads(file_path.read_text())
+            matching_models = [
+                model_values
+                for model, model_values in content["nodes"].items()
+                if model.split(".")[-1] == parts[-1]
+            ]
+            if matching_models:
+                return [col.lower() for col in matching_models[0]["columns"].keys()]
```

Comment:
We can use the `self.column_casing(col)`
Its basically a function which looks at the db adapter and if it is snowflake, it manages casing. Its not ideal but it at least gives us a single method to hack at to better manage how we do column casing. 

LMK if this makes sense. We should probably have our own dedicated `Column` class eventually designed specifically for comparisons.